### PR TITLE
enhancement(external docs): Add Vector config examples to main page of vector.dev

### PR DESCRIPTION
**Pull request by lucperkins**
This PR also fixes some typos.

### CODE REVIEW

File: docs/content/en/_index.md [hunk 1]
```diff
@@ -48,47 +48,51 @@ configure:
...
+      [sources.kafka_in]
+      type = "kafka"
+      bootstrap_servers = "10.14.22.123:9092,10.14.23.332:9092"
+      group_id = "vector-logs"
+      message_key = "message"
+      topics = ["logs-*"]
+
+      [sinks.elasticsearch_out]
+      type = "elasticsearch"
+      inputs = ["kafka_in"]
```
Comment (spencergilbert) [inline docs/content/en/_index.md:80]: Technically I think we need to `parse_json` before sending across to ES

File: docs/content/en/_index.md [hunk 2]
```diff
@@ -48,47 +48,51 @@ configure:
...
-      type = "console"
-      encoding.codec = "text"
-  - title: And yet another
+  - title: Kubernetes logs to AWS S3
     config: |
-      [sources.and_yet_another]
-      type = "other"
+      [sources.k8s_in]
+      type = "kubernetes_logs"
+      exclude_paths_glob_patterns = []
```
Comment (spencergilbert) [inline docs/content/en/_index.md:88]: Is there a reason you've added the glob pattern option?

File: docs/content/en/_index.md (hunk 1 above)
Comment (lucperkins) [reply docs/content/en/_index.md]: Okay, I updated this. Let me know if the new version works. I'm a bit rusty with configs.

File: docs/cue/reference/examples.cue [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			type = "kafka"
+			bootstrap_servers = "10.14.22.123:9092,10.14.23.332:9092"
+			group_id = "vector-logs"
+			message_key = "message"
+			topics = ["logs-*"]
+
+			[transforms.json_parse]
+			type = "remap"
+			source = '''
+			  . = parse_json!(.message)
```
Comment (spencergilbert) [inline docs/cue/reference/examples.cue:42]:
```suggestion
			  . |= parse_json!(.message)
```

Comment (spencergilbert) [reply docs/cue/reference/examples.cue:42]: Unless we want to be wordier: `. = merge!(parse_json!(.message))`

**Review by spencergilbert — APPROVED** [review]

File: docs/cue/reference/examples.cue [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			bootstrap_servers = "10.14.22.123:9092,10.14.23.332:9092"
+			group_id = "vector-logs"
+			message_key = "message"
+			topics = ["logs-*"]
+
+			[transforms.json_parse]
+			type = "remap"
+			source = '''
+			  . = merge!(parse_json!(.message))
+			'''
```
Comment (spencergilbert) [inline docs/cue/reference/examples.cue:43]:
```suggestion
			[transforms.json_parse]
			type = "remap"
			source = '''
			parsed, err = parse_json!(.message)
			if err != null {
			  log(err, level: "error")
			}
			  . = merge(parsed ?? {})
			'''
```

cc @jszwedko talked to @lucperkins about showing "best practices" in our examples - does this track with your expectations?

Comment (lucperkins) [reply docs/cue/reference/examples.cue:43]: @spencergilbert The `!` can be removed here, right?

Comment (jszwedko) [reply docs/cue/reference/examples.cue]: Yeah, I like showing the error handling as "best practice". You should remove the `!` there.

Comment (jszwedko) [reply docs/cue/reference/examples.cue]: And the last bit should be `. |= parsed ?? {}` or `. = merge(., parsed ?? {})`.

Comment (spencergilbert) [reply docs/cue/reference/examples.cue]: Er, yea - I think a bad copy + paste

Comment (spencergilbert) [reply docs/cue/reference/examples.cue]: > And the last bit should be `. |= parsed ?? {}` or `. = merge(., parsed ?? {})`.

LOL - I'm bad at things

File: docs/cue/reference/examples.cue [hunk 5]
```diff
@@ -0,0 +1,84 @@
+package metadata
```
Comment (jszwedko) [inline docs/cue/reference/examples.cue:1]: Nice 👍 

It should be possible to wire these up with the tests in `lib/vrl/tests` to ensure they run and don't break if you feel inspired.

File: docs/cue/reference/examples.cue [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+			"""#
+	},
+	{
+		title: "Kafka topic to Elasticsearch"
+		example: #"""
+			[sources.kafka_in]
+			type = "kafka"
+			bootstrap_servers = "10.14.22.123:9092,10.14.23.332:9092"
+			group_id = "vector-logs"
+			message_key = "message"
```
Comment (jszwedko) [inline docs/cue/reference/examples.cue:36]: This is an invalid key, it should be `key_field'.

File: docs/cue/reference/examples.cue [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+	example: !=""
+}
+
+config_examples: [#ConfigExample, ...#ConfigExample] & [
+	{
+		title: "Redacted Datadog Agent logs to Datadog"
+		example: #"""
+			[sources.datadog_agent]
+			type = "datadog_logs"
+			address = "localhost:80"
```
Comment (jszwedko) [inline docs/cue/reference/examples.cue:14]: You need to use an IP address here, not a hostname.

File: docs/cue/reference/examples.cue [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+			index = "logs-via-kafka"
+			"""#
+	},
+	{
+		title: "Kubernetes logs to AWS S3"
+		example: #"""
+			[sources.k8s_in]
+			type = "kubernetes_logs"
+
+			[sinks.aws_s3_out]
```
Comment (jszwedko) [inline docs/cue/reference/examples.cue:62]: This is missing `encoding.codec`

File: docs/cue/reference/examples.cue [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+			region = "us-east-1"
+			compression = "gzip"
+			"""#
+	},
+	{
+		title: "Splunk HEC to Datadog"
+		example: #"""
+			[sources.splunk_hec_in]
+			type = "splunk_hec"
+			address = "localhost:8080"
```
Comment (jszwedko) [inline docs/cue/reference/examples.cue:75]: This needs to be an IP address.

**Review by jszwedko — COMMENTED** [review]
@lucperkins some of these examples are invalid. We should make sure they are runnable 😄

File: docs/cue/reference/examples.cue (hunk 7 above)
Comment (jszwedko) [reply docs/cue/reference/examples.cue:14]: I'd suggest `0.0.0.0` for these.

File: docs/cue/reference/examples.cue (hunk 8 above)
Comment (jszwedko) [reply docs/cue/reference/examples.cue:62]: I'd suggest `encoding.codec = "json"`

File: docs/cue/reference/examples.cue (hunk 5 above)
Comment (lucperkins) [reply docs/cue/reference/examples.cue:1]: Hmmm. As far as I know we don't have the facilities to test full end-to-end Vector configs like this. I was thinking, though, that I might want to add a separate VRL section to the main page, and I do think the VRL tests could be wired up to that.

Comment (jszwedko) [reply docs/cue/reference/examples.cue:1]: Yeah, you are right. I neglected these were full configs, not just VRL snippets. We could at least run them through `vector validate --no-environment` though to catch simple errors.

**Pull request merged**